### PR TITLE
Add consts to cmdline code

### PR DESCRIPTION
**Pull request by yashykt**

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

File: include/grpc/support/cmdline.h [hunk 1]
```diff
@@ -75,7 +75,8 @@ GPRAPI void gpr_cmdline_set_survive_failure(gpr_cmdline* cl);
 /** Parse the command line; returns 1 on success, on failure either dies
    (by default) or returns 0 if gpr_cmdline_set_survive_failure() has been
    called */
-GPRAPI int gpr_cmdline_parse(gpr_cmdline* cl, int argc, char** argv);
+GPRAPI int gpr_cmdline_parse(gpr_cmdline* cl, int argc,
+                             const char* const* argv);
```
Comment (vjpai) [inline include/grpc/support/cmdline.h]: `include/grpc` needs to be pure C89

**Comment by vjpai** [comment]
Ignore my last review; I thought C89 didn't allow const. Re-reviewing.

File: src/core/lib/support/cmdline.cc [hunk 2]
```diff
@@ -237,18 +237,18 @@ static int value_state(gpr_cmdline* cl, char* str) {
       }
       break;
     case ARGTYPE_STRING:
-      *(char**)cl->cur_arg->value = str;
+      *(const char**)cl->cur_arg->value = str;
```
Comment (vjpai) [inline src/core/lib/support/cmdline.cc]: Avoid C-style cast.

File: src/core/lib/support/cmdline.cc [hunk 3]
```diff
@@ -287,9 +287,10 @@ static int normal_state(gpr_cmdline* cl, char* str) {
     eq = strchr(str, '=');
     if (eq != nullptr) {
       /* copy the string into a temp buffer and extract the name */
-      tmp = arg_name = (char*)gpr_malloc((size_t)(eq - str + 1));
-      memcpy(arg_name, str, (size_t)(eq - str));
-      arg_name[eq - str] = 0;
+      tmp = (char*)gpr_malloc((size_t)(eq - str + 1));
```
Comment (vjpai) [inline src/core/lib/support/cmdline.cc]: Avoid C-style cast.

**Comment by yashykt** [comment]
Addressed the comments.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by ctiller** [comment]
Correct thing to do is to use gflags and delete cmdline.{h,cc} entirely...
It was written so we had a command line parser in C for testing, and that
reason to exist is now voided.

**Comment by vjpai** [comment]
@ctiller : excellent point and now is a good time to kill it since there are already core API changes inflight.

**Comment by yashykt** [comment]
Since it is in public API, it is possible for it being used legally. Do we still want to remove it altogether? Or start using gflags in our test suite and keep this as legacy API for the timebeing?

**Comment by ctiller** [comment]
Bump the core version number next release (this is happening anyway) and
delete it.

I'll wager nobody picked up gRPC for its C89 command line parser... Though
I thought it was quite good for the code size.

**Comment by yashykt** [comment]
Create PR #13562 to remove gpr_cmdline

**Pull request closed without merging**